integration_tests/pages/apply/taskListPage.ts

```diff
@@ -1,10 +1,24 @@
 import Page from '../page'
+import Apply from '../../../server/form-pages/apply'
 
 export default class TaskListPage extends Page {
   constructor() {
     super('CAS 2: Refer for Accommodation')
   }
 
+  shouldShouldSectionsAndTasks = (): void => {
+    Apply.sections.forEach((section, sectionIndex) => {
+      cy.get('.app-task-list__section').contains(section.title)
+      section.tasks.forEach((task, taskIndex) => {
+        if (sectionIndex === 0 && taskIndex === 0) {
+          cy.get('h2').contains(section.title).siblings('ul').find('.app-task-list__task-name').contains(task.title)
+        } else {
+          cy.get('h2').contains(section.title).siblings('ul').find('.app-task-list__task-name').contains(task.title)
+        }
+      })
+    })
+  }
+
```

Comment:
This test helper could perhaps be clearer? Two suggestions:

### 1. Naming of helper
A more descriptive name to make it clear that it's asserting that each section contains the expected tasks. Something like `shouldShowTasksWithinTheirSections()` maybe?

### 2. More readable assertion code
Probably we can make the code in the assertion more readable and maintainable? It was the fiddly nature of making this assertion about the grouping of tasks into sections which prompted me to add the `data-section_name` attribute to each section `<li>` (see https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/commit/0c450e2cbfd464fa29f88d1fc6fbb25fc7ca485d). Perhaps we could use that simpler style of code (leveraging the `within()` function to describe grouping) here in your general purpose assertion, something like:

```
shouldShowTasksWithinTheirSections = (): void => {
    Apply.sections.forEach(section => {
      cy.get(`[data-section_name="${section.title}"]`).within(() => {
        // And I see the expected SECTION
        cy.get('.app-task-list__section').contains(section.title)

        // And I see each expected TASK
        section.tasks.forEach(task => {
          cy.get('.app-task-list__task-name').contains(task.title)
        })
      })
    })
  }
```

TIDY UP
We should probably remove the now unused helpers:

- `shouldShowAreaAndFundingSection()` and
- `shouldShowAboutPersonSection()`

Reply:
yes, agree that is a significant improvement, have implemented